lib/matplotlib/cbook/__init__.py

```diff
@@ -678,33 +678,46 @@ def clear(self):
 
     def bubble(self, o):
         """
-        Raise *o* to the top of the stack.  *o* must be present in the stack.
+        Raise *o* to the top of the stack and return it.
 
-        *o* is returned.
+        If *o* is contained multiple times, all references are moved to the
+        top of the stack.
+
+        Raises
+        ------
+        ValueError
+            If *o* is not in the stack.
         """
         if o not in self._elements:
-            raise ValueError('Unknown element o')
-        old = self._elements[:]
+            raise ValueError('Given element not contained in the stack.')
+        old_elements = self._elements.copy()
         self.clear()
-        bubbles = []
-        for thiso in old:
-            if thiso == o:
-                bubbles.append(thiso)
+        top_elements = []
+        for elem in old_elements:
+            if elem == o:
+                top_elements.append(elem)
             else:
-                self.push(thiso)
-        for _ in bubbles:
+                self.push(elem)
+        for _ in top_elements:
             self.push(o)
         return o
```

Comment:
Shorter would be something like
```
count = self._elements.count(o)
if not count: ...
self._elements = [e for e in self._elements if e != o] + n * [o]
```
no?

Reply:
That's what I meant with "should be tested". I believe that there are subtle differences if you have objects that are equal but not identical. Not that I think the sematics for that is considered in the design and well defined, but I don't want to change this without really thinking about it.